src/Component/src/State/Provider/DeserializeProvider.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\State\Provider;
+
+use Sylius\Resource\Context\Context;
+use Sylius\Resource\Context\Option\RequestOption;
+use Sylius\Resource\Metadata\DeleteOperationInterface;
+use Sylius\Resource\Metadata\HttpOperation;
+use Sylius\Resource\Metadata\Operation;
+use Sylius\Resource\State\ProviderInterface;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class DeserializeProvider implements ProviderInterface
+{
+    public function __construct(
+        private ProviderInterface $decorated,
+        private ?SerializerInterface $serializer,
+    ) {
+    }
+
+    public function provide(Operation $operation, Context $context): object|array|null
+    {
+        $data = $this->decorated->provide($operation, $context);
+
+        // We need request content
```

Comment:
This comment message doesn't help me understand anything, so I would vote for removal. 

Overall it's to easy to read that provider checks if operation is deserializeable and either returns inner provider data or deserialize it.